### PR TITLE
(feat) O3-1809: Using global Visit setting to either map Order Encounter to Active Visit

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR uses the global visit setting to either map Order Encounter to current Visit or add to the current day's encounter.

Currently, the order basket mandates for the patient to have an Active Visit, or else the order basket doesn't opens up.
This PR will unblock the distributions not using the visits. 

## Related Issue
https://issues.openmrs.org/browse/O3-1809

### CODE REVIEW

File: packages/esm-patient-medications-app/src/config-schema.ts [hunk 1]
```diff
@@ -45,4 +50,5 @@ export interface ConfigObject {
     display: string;
   };
   drugOrderTypeUUID: string;
+  mapOrderEncounterToCurrentVisit: boolean;
```
Comment (samuelmale) [inline packages/esm-patient-medications-app/src/config-schema.ts:53]: Is this typing meant for `config.currentVisitRequired`? If so we need to align on the naming.

**Review by samuelmale — COMMENTED** [review]
I know this is a work in progress; just an early review 😉 

Hope you saw the build failure..

File: packages/esm-patient-medications-app/src/medications-summary/order-basket-action-button.test.tsx [hunk 2]
```diff
@@ -9,9 +9,18 @@ import {
 } from '@openmrs/esm-patient-common-lib';
 import { mockPatient } from '../../../../__mocks__/patient.mock';
 import OrderBasketActionButton from './order-basket-action-button.component';
+import { useSystemVisitSetting } from '../api/api';
 
 const mockedUseLayoutType = useLayoutType as jest.Mock;
 const mockUsePatient = usePatient as jest.Mock;
+useSystemVisitSetting;
```
Comment (samuelmale) [inline packages/esm-patient-medications-app/src/medications-summary/order-basket-action-button.test.tsx:16]: Do you need this?

File: packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx [hunk 3]
```diff
@@ -53,6 +58,7 @@ const OrderBasket = connect<OrderBasketProps, OrderBasketStoreActions, OrderBask
     isLoading: isLoadingOrders,
     isValidating,
   } = usePatientOrders(patientUuid, 'ACTIVE');
+  const noCurrentVisit = activeVisitRequired && !currentVisit;
```
Comment (samuelmale) [inline packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx:61]: What happens if `activeVisitRequired` is `false` and `currentVisit` is truthy?

Comment (samuelmale) [inline packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx:61]: What happens if `activeVisitRequired` is `false` and `currentVisit` is truthy?

File: packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx [hunk 4]
```diff
@@ -95,12 +101,12 @@ const OrderBasket = connect<OrderBasketProps, OrderBasketStoreActions, OrderBask
       orderEncounterUuid = await createEmptyEncounter(
         patientUuid,
         config?.drugOrderEncounterType,
-        currentVisit?.uuid,
+        activeVisitRequired ? currentVisit?.uuid : null,
```
Comment (samuelmale) [inline packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx:104]: What happens if we have an existing visit but `activeVisitRequired` is `false`?

File: packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx (hunk 3 above)
Comment (vasharma05) [reply packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx:61]: If the activeVisit is not required, then we don't handle the current visit at all.

But the important point here will be if the Visits are disabled from the server, why will there be a current visit.

File: packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx (hunk 4 above)
Comment (vasharma05) [reply packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx:104]: `activeVisitRequired` is the global setting in the legacy UI which the admin has already disabled, and hence we don't consider visits at all.

File: packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx (hunk 3 above)
Comment (samuelmale) [reply packages/esm-patient-medications-app/src/order-basket/order-basket.component.tsx:61]: Makes sense

File: packages/esm-patient-medications-app/src/medications-summary/order-basket-action-button.test.tsx (hunk 2 above)
Comment (vasharma05) [inline packages/esm-patient-medications-app/src/medications-summary/order-basket-action-button.test.tsx:16]:
```suggestion

```

**Review by samuelmale — APPROVED** [review]

**Comment by samuelmale** [comment]
@vasharma05 / @denniskigen any idea as to why the "Bundle size" job is failing?

**Comment by vasharma05** [comment]
> @vasharma05 / @denniskigen any idea as to why the "Bundle size" job is failing?

As per @denniskigen, there's a node v18 conflict with the angular package. The CI/CD job needs to be updated.
@denniskigen , any update on this?
Thanks

**Comment by denniskigen** [comment]
Yeah, @vasharma05, I've fixed the issue. Rebase against main and you should be good to go.

**Comment by vasharma05** [comment]
Thanks a lot @denniskigen !

**Pull request merged**